dask_match/core.py

```diff
@@ -307,20 +313,32 @@ def _divisions(self):
     def _name(self):
         return funcname(self.operation) + "-" + tokenize(*self.operands)
 
-    def _layer(self):
+    def _subgraph_dependencies(self):
+        # List `Expr` operands only.
+        # This property is required to enable fusion
+        return [operand for operand in self.operands if isinstance(operand, Expr)]
+
+    def _block_subgraph(self):
         return {
-            (self._name, i): (
+            self._name: (
                 apply,
                 self.operation,
                 [
-                    (operand._name, i) if isinstance(operand, Expr) else operand
+                    operand._name if isinstance(operand, Expr) else operand
                     for operand in self.operands
                 ],
                 self._kwargs,
             )
-            for i in range(self.npartitions)
         }
 
+    def _layer(self):
+        return _subgraph_callable_layer(
```

Comment:
Maybe this method could be inlined here?  I think that we only need to use this logic in this one place.

Reply:
I was hesitant to in-line this because we are also using it for `Fusable` expressions that don't inherit from `Blockwise` (i.e. `ReadParquet` and `FromPandas`). I realize we can make `IO` a `Blockwise` subclass, but that will require a bit more work in` ReadCSV`.  Also, I wasn't sure if we actually **want** to require all "fuseable" expressions to be `Blockwise` subclasses, though it is certainly fine with me if we do.

Reply:
Decided it was worthwhile to make everything "fusable" be a `Blockwise` subclass, so this is now inlined.